### ISSUE TITLE
Weird behaviour

### ISSUE THREAD

**Issue by moufmouf** [thecodingmachine/workadventure#208]
Extremely weird behavior in one given instance of a room:
![image](https://user-images.githubusercontent.com/1290952/86028962-ddc49300-ba32-11ea-93d0-6c60a7ff2e74.png)

When connecting, a group is directly joined where I am alone.
Other players in the same room see the same behavior. I cannot see them, they cannot see me but I can see the circle of their group.
When they move, errors are displayed in the console (about players not found)

In the logs, we can see error on "join_room".

[log_wa.txt](https://github.com/thecodingmachine/workadventure/files/4846622/log_wa.txt)

**Comment by moufmouf** [comment]
More logs:

```
An error occurred on "join_room" event
Error: Could not find user with id 14d51cea-6cf8-44a6-8b02-f3921ecde4a8
    at IoSocketController.searchClientByIdOrFail (/usr/src/app/dist/src/Controller/IoSocketController.js:254:19)
    at /usr/src/app/dist/src/Controller/IoSocketController.js:163:44
    at Function.from (<anonymous>)
    at Socket.<anonymous> (/usr/src/app/dist/src/Controller/IoSocketController.js:161:45)
    at Socket.emit (events.js:315:20)
    at /usr/src/app/node_modules/socket.io/lib/socket.js:528:12
    at processTicksAndRejections (internal/process/task_queues.js:79:11)
An error occurred on "user_position" event
Error: Could not find user with id 14d51cea-6cf8-44a6-8b02-f3921ecde4a8
    at IoSocketController.searchClientByIdOrFail (/usr/src/app/dist/src/Controller/IoSocketController.js:254:19)
    at IoSocketController.connectedUser (/usr/src/app/dist/src/Controller/IoSocketController.js:361:27)
    at Group.connectCallback (/usr/src/app/dist/src/Controller/IoSocketController.js:284:23)
    at Group.join (/usr/src/app/dist/src/Model/Group.js:47:14)
    at /usr/src/app/dist/src/Model/Group.js:12:19
    at Array.forEach (<anonymous>)
    at new Group (/usr/src/app/dist/src/Model/Group.js:11:15)
    at World.updatePosition (/usr/src/app/dist/src/Model/World.js:56:33)
    at Socket.<anonymous> (/usr/src/app/dist/src/Controller/IoSocketController.js:189:27)
    at Socket.emit (events.js:315:20)
An error occurred on "disconnect"
Error: Could not find group
    at World.leaveGroup (/usr/src/app/dist/src/Model/World.js:93:23)
    at World.leave (/usr/src/app/dist/src/Model/World.js:35:18)
    at IoSocketController.leaveRoom (/usr/src/app/dist/src/Controller/IoSocketController.js:265:23)
    at Socket.<anonymous> (/usr/src/app/dist/src/Controller/IoSocketController.js:215:27)
    at Socket.emit (events.js:315:20)
    at Socket.emit (/usr/src/app/node_modules/socket.io/lib/socket.js:142:10)
    at Socket.onclose (/usr/src/app/node_modules/socket.io/lib/socket.js:454:8)
    at Socket.ondisconnect (/usr/src/app/node_modules/socket.io/lib/socket.js:418:8)
    at Socket.onpacket (/usr/src/app/node_modules/socket.io/lib/socket.js:340:12)
    at /usr/src/app/node_modules/socket.io/lib/client.js:214:16
```

When connecting a new user, the list of already connected users to this room should be sent. The fact is one of those users stored in `world.getUsers()` does not exist anymore in `this.sockets`. So it is like if when a user disconnected, the user was not remove from the group for some reason (but there is no trace of that in the logs). (or if the disconnect event was never sent by socket.io....)

**Comment by moufmouf** [comment]
Side note: in IOSocketController, `this.Worlds` is never emptied!

**Comment by moufmouf** [comment]
> Side note: in IOSocketController, this.Worlds is never emptied!

Closed by #209
